File: Components/shared/src/main/se/clouds/javanet/app/domain/model/SharedFeature.java
package se.clouds.javanet.app.domain.model;

import java.util.HashMap;
import java.util.Map;

public class SharedFeature
{
    public String Name;
    public String Value;
    public boolean Activated;
    public String Description;

    public SharedFeature(String name, boolean activated, String value, String description) {
        Name = name;
        Activated = activated;
        Value = value;
        Description = description;
    }

    public String getName() {
        return Name;
    }

    public void setName(String name) {
        Name = name;
    }

    public boolean isActivated() {
        return Activated;
    }

    public void setActivated(boolean activated) {
        Activated = activated;
    }

    public String getDescription() {
        return Description;
    }

    public void setDescription(String description) {
        Description = description;
    }

    public Map<String, Object> AsMap(){
        Map<String, Object> map = new HashMap<String, Object>();
        map.put("name", Name);
        map.put("activated", Activated);
        map.put("value", Value);
        map.put("description", Description);
        return map;
    }
    public static SharedFeature FromMap(Map<String, Object> map)
    {
        return new SharedFeature(
            (String)map.get("name"),
            (boolean)map.get("activated"),
            (String)map.get("value"),
            (String)map.get("description"));
    }
}

File: Components/shared/src/main/se/clouds/javanet/app/domain/model/SharedFeature.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;
using se.clouds.javanet.core.domain;

namespace se.clouds.javanet.app.domain.model
{
    public class SharedFeature : AggregateRoot<SharedFeature>
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Key]
        [JsonIgnore]
        public Guid Id { get; set; }

        public bool Activated { get; set; }

        public string Value { get; set; }
        public string Version { get; set; }

        public string Description { get; set; }


        public override string ToString()
        {
            return string.Format($"Feature (Name = {Name}, Activated = {Activated}, Value = {Value}, Description = {Description})");
        }

        protected override bool EqualsCore(SharedFeature other)
        {
            throw new NotImplementedException();
        }

        protected override int GetHashCodeCore()
        {
            throw new NotImplementedException();
        }
    }
}
